security/interceptor/security-interceptor-anonymous-wrapper/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptorWrapper.java

```diff
@@ -37,14 +42,26 @@ public AnonymousInterceptorWrapper() {
         //make sure this interceptor runs before the WSS4J one in the same Phase, otherwise it won't work
         Set<String> before = getBefore();
         before.add(WSS4JInInterceptor.class.getName());
+        before.add(PolicyBasedWSS4JInInterceptor.class.getName());
 	}
 
-    public void setAnonIntercep(PhaseInterceptor anonIntercep) {
-        this.anonIntercep = anonIntercep;
+    protected BundleContext getContext() {
+        Bundle cxfBundle = FrameworkUtil.getBundle(AnonymousInterceptorWrapper.class);
+        if (cxfBundle != null) {
+            return cxfBundle.getBundleContext();
+        }
+        return null;
     }
 
     @Override
 	public void handleMessage(SoapMessage msg) throws Fault {
+        BundleContext context = getContext();
+        PhaseInterceptor anonIntercep = null;
+        ServiceReference anonIntercepRef = context.getServiceReference(PhaseInterceptor.class.getName());
+
+        if (anonIntercepRef != null) {
+            anonIntercep = (PhaseInterceptor) context.getService(anonIntercepRef);
```

Comment:
Could be null or throw some exception. Should probably protect against that.
https://osgi.org/javadoc/r4v43/core/org/osgi/framework/BundleContext.html#getService(org.osgi.framework.ServiceReference)